wikipedia_cleanup/data_processing.py

```diff
@@ -56,45 +64,126 @@ def read_pickle_file(file_path: Path) -> List[InfoboxChange]:
         return pickle.load(file)  # type: ignore
 
 
-def read_file(file_path: Path) -> List[InfoboxChange]:
+def sort_changes(changes: List[InfoboxChange]) -> List[InfoboxChange]:
+    changes.sort(
+        key=lambda change: (
+            change.page_id,
+            change.infobox_key,
+            change.property_name,
+            change.value_valid_from,
+        )
+    )
+    return changes
+
+
+def read_file_sorted(file_path: Path) -> List[InfoboxChange]:
     if "pickle" in str(file_path):
+        # pickle is already sorted
         return read_pickle_file(file_path)
     elif "json" in str(file_path):
```

Comment:
file path should probably end with those. otherwise `test_pickle.json` might give us problems. `Path` has a function for data type as far as i know.

Reply:
`file_path.suffix == '.pickle'` should work